modules/run-vault/run-vault

```diff
@@ -282,58 +286,79 @@ listener "tcp" {
 }\n
 EOF
 )
-
-  local consul_storage_type="storage"
-  local dynamodb_storage_type="storage"
-  local s3_config=""
   local vault_storage_backend=""
-  if [[ "$enable_s3_backend" == "true" ]]; then
-    s3_config=$(cat <<EOF
+  local vault_ha_storage_backend=""
+  if [[ "$storage_backend" == "consul" ]]; then
```

Comment:
NIT: to keep this function from growing bigger and bigger, we should extract the code that sets the storage backend into a separate function that uses a case statement and returns the proper value by writing it to stdout. Same for HA storage.